eva/expression/function_expression.py

```diff
@@ -91,11 +92,13 @@ def evaluate(self, batch: Batch, **kwargs) -> Batch:
         return outcomes
 
     def _gpu_enabled_function(self):
-        if isinstance(self._function, GPUCompatible):
-            device = self._context.gpu_device()
-            if device != NO_GPU:
-                return self._function.to_device(device)
-        return self._function
+        if self._function_instance is None:
```

Comment:
@xzdandy @kaushikravichandran Is this piece required to avoid multiple instantiations of the same model?

Reply:
Yes. This optimization is valid even without ray. If you recall the cprofile we were doing, this is one significant source of overhead.